### DISCUSSION TITLE
Background location tracking

### DISCUSSION THREAD

**Discussion by kamalsprasad** [quasarframework/quasar#12579] [CLI - Capacitor / Cordova modes] [answered]
Hi,

Does anyone have any suggestions for (capacitor based) background location tracking for Quasar? I found this possible solution but am having difficulty in getting it working in my Android app. https://github.com/capacitor-community/background-geolocation I have the watcher in my App.vue file. Should it be somewhere else?

TIA,

Kamal

**Comment by MilosPaunovic** [comment, marked as answer]
There is an example of using Capacitor’s `Geolocation` API on the Quasar docs; have you tried checking that out?
[https://quasar.dev/quasar-cli/developing-capacitor-apps/capacitor-api#example-geolocation](https://quasar.dev/quasar-cli/developing-capacitor-apps/capacitor-api#example-geolocation)

**Reply by kamalsprasad** [reply]
I did get it working with https://github.com/capacitor-community/background-geolocation by creating a new app just to configure that and transferred the code to my main application for which I had posted this question.

**Comment by kamalsprasad** [comment]
Thanks for your reply, Milos. Yes, I've tried that code but using the watcher in App.vue seems to break the Geolocation.getCurrentLocation() elsewhere in the app. E.g. If I try to use Geolocation.getCurrentLocation() in Map.vue component when I have a watcher in my App.vue, it doesn't work. Maybe I'm doing something wrong. I'm learning as I'm going so I appreciate your advice.

**Reply by MilosPaunovic** [reply]
My best advice is to create a minimal reproduction repository for someone to be able to play with it and help you find a solution.

**Reply by kamalsprasad** [reply]
@MilosPaunovic, that was really good advice. I created an app just with the capacitor community plugin I listed above and was able to figure out exactly what needed to be done.  Please suggest an answer and I'll mark it as an answer. Thanks so much!!

**Reply by MilosPaunovic** [reply]
You can mark the one from above as the answer, as it's always a good practice to consult official docs first.
